modules/javafx.graphics/src/main/java/com/sun/glass/ui/MoveResizeHelper.java

```diff
@@ -0,0 +1,199 @@
+        RESIZE_CURSORS.put(WindowRegion.RIGHT, application.createCursor(Cursor.CURSOR_RESIZE_LEFTRIGHT));
+        RESIZE_CURSORS.put(WindowRegion.TOP, application.createCursor(Cursor.CURSOR_RESIZE_UPDOWN));
+        RESIZE_CURSORS.put(WindowRegion.BOTTOM, application.createCursor(Cursor.CURSOR_RESIZE_UPDOWN));
+    }
+
+    private final View view;
+    private final Window window;
+    private final WindowRegionClassifier regionClassifier;
+    private int mouseDownX, mouseDownY;
+    private int mouseDownWindowX, mouseDownWindowY;
+    private int mouseDownWindowWidth, mouseDownWindowHeight;
+    private WindowRegion currentWindowRegion;
+    private Cursor lastCursor;
+
+    public MoveResizeHelper(View view, Window window) {
+        this.view = view;
+        this.window = window;
+        this.regionClassifier = window.regionClassifier;
+    }
+
+    public final boolean handleMouseEvent(int type, int button, int x, int y, int xAbs, int yAbs) {
+        int wx = (int)(x / window.getPlatformScaleX());
+        int wy = (int)(y / window.getPlatformScaleY());
+
+        if (type != MouseEvent.DRAG) {
+            var eventHandler = view.getEventHandler();
+            Node pickedNode = eventHandler != null ? eventHandler.pickNode(wx, wy) : null;
+            currentWindowRegion = regionClassifier.classify(wx, wy, pickedNode);
+            updateCursor(currentWindowRegion);
+
+            if (currentWindowRegion == WindowRegion.CLIENT) {
+                return false;
+            }
+        }
+
+        switch (type) {
+            case MouseEvent.DRAG:
+                handleMouseDrag(button, xAbs, yAbs, currentWindowRegion);
+                break;
+            case MouseEvent.DOWN:
+                handleMouseDown(xAbs, yAbs);
+                break;
+        }
+
+        return false;
+    }
+
+    protected boolean shouldStartMoveDrag(int button, WindowRegion region) {
+        return button == MouseEvent.BUTTON_LEFT && region == WindowRegion.TITLE;
+    }
+
+    protected boolean shouldStartResizeDrag(int button, WindowRegion region) {
+        return button == MouseEvent.BUTTON_LEFT && RESIZE_CURSORS.get(region) != null;
+    }
+
+    private void handleMouseDrag(int button, int xAbs, int yAbs, WindowRegion region) {
+        if (shouldStartMoveDrag(button, region)) {
+            handleMoveWindow(xAbs, yAbs);
+        } else if (shouldStartResizeDrag(button, region)) {
+            handleResizeWindow(xAbs, yAbs, region);
+        }
+    }
+
+    private void handleMouseDown(int xAbs, int yAbs) {
+        mouseDownX = xAbs;
+        mouseDownY = yAbs;
+        mouseDownWindowX = window.getX();
+        mouseDownWindowY = window.getY();
+        mouseDownWindowWidth = window.getWidth();
+        mouseDownWindowHeight = window.getHeight();
+    }
+
+    private void handleMoveWindow(int xAbs, int yAbs) {
+        window.setPosition(mouseDownWindowX + xAbs - mouseDownX, mouseDownWindowY + yAbs - mouseDownY);
+    }
```

Comment:
On Linux this will be constrained by the window manager, so it's not possible to move the window outside of the desktop bounds, including panels.

This can be done by implementing a `beginMoveDrag` on glass that will call (on linux):
`gdk_window_begin_move_drag`